src/cli/commands/check.js

```diff
@@ -40,7 +40,7 @@ export async function run(
   }
 
   // get patterns that are installed when running `yarn install`
-  const [, rawPatterns] = await install.hydrate();
+  const [, rawPatterns] = await install.hydrate(true);
```

Comment:
As a matter of fact, I think there should be a third mode for check - https://github.com/yarnpkg/yarn/issues/1934.
I would rather rely on that and the integrity check than the default one